projects/frontend/cicd/test_e2e_data_pipelines.sh

```diff
@@ -0,0 +1,31 @@
+#!/bin/bash -e
+
+# Copyright 2021-2023 VMware, Inc.
+# SPDX-License-Identifier: Apache-2.0
+
+###
+# For installing the build prerequisites, consider `install_data_pipelines.sh` script.
+# To rebuild, link and run unit tests, consider `build_data_pipelines.sh`.
+# This script is meant to run the e2e tests of data-pipelines component.
+###
+
+if ! which npm >/dev/null 2>&1 ; then
+  echo "ERROR:"
+  echo "Please install npm 8.5.5+. Build cannot continue without it."
+  exit 1
+fi
+echo "Logging npm engines version..."
+npm version
+
+cd "$(dirname $0)" || exit 1
+cd "../data-pipelines/gui"
+
+if [ $# -eq 0 ]
+  then
+    echo "ERROR: No OAuth2 token provided."
+    exit 3
+fi
+export OAUTH2_API_TOKEN=$1
```

Comment:
Instead of passing the API token as an argument to the script, I think it's better to use environment variables to store sensitive information. The environment variables can be set in the GitLab CI/CD settings and then accessed within the script. 

A few reasons : 
 * Environment variables are more secure than arguments because they are not visible in the process list or command history. 
 * Avoiding clutter in the future - As a script grows in complexity, it can become difficult to manage and maintain multiple command line arguments.

Reply:
The proposal made, makes quite less likely for the (dev local setup) secret to be copied and shared by accident. That applies for a usage, where a user pasted the parameter directly. 

Not sure an environment-scoped secret (broader visibility) than a parameter (local scope) is more secure, due longer life-cycle and global accessibility. In case of parameter, the user may decide to source it in different ways (including env var). 
Some examples and analysis made https://smallstep.com/blog/command-line-secrets/ 

Do you think we need to transition?

Reply:
I think env. variables are a bit more secure than command-line arguments. But we could do better. 
I do also think that they are more maintainable (in the same way that keyword arguments are easier to maintain than positional arguments) 

But we've used command line arguments before (though I've never recommended it) and we can continue using it and re-evaluate in separate change (the focus of this change is different).